source/hwIo.py

```diff
@@ -178,9 +177,19 @@ def close(self):
 
 	def _notifyReceive(self, data):
 		# Set the timeout for onReceive in case it does a sync read.
-		self._ser.timeout = self._origTimeout
+		self._setTimeout(self._origTimeout)
 		super(Serial, self)._notifyReceive(data)
-		self._ser.timeout = None
+		self._setTimeout(None)
+
+	def _setTimeout(self, timeout):
+		# #6035: pyserial reconfigures all settings of the port when setting a timeout.
+		# This can cause error 'Cannot configure port, some setting was wrong.'
+		# Therefore, manually set the timeout using the Win32 API.
+		# timeout=None to clear all timeouts, else a positive number of milliseconds.
+		timeouts = COMMTIMEOUTS()
+		if timeout is not None:
+			timeouts.ReadTotalTimeoutConstant = max(int(timeout * 1000), 1)

```

Comment:
While this certainly sets the read timeout, unless I'm missing something, it also resets all other timeouts to 0 (including the write timeout). We can't assume the caller is only interested in the read timeout. Drivers do set the write timeout. In fact, no write timeout could be quite bad for real serial ports, since they may have no way of knowing the device was disconnected.